packages/flutter/test/services/system_chrome_test.dart

```diff
@@ -89,16 +89,64 @@ void main() {
     ));
   });
 
+  test('setEnabledSystemUIMode control test', () async {
+    final List<MethodCall> log = <MethodCall>[];
+
+    TestDefaultBinaryMessengerBinding.instance!.defaultBinaryMessenger.setMockMethodCallHandler(SystemChannels.platform, (MethodCall methodCall) async {
+      log.add(methodCall);
+    });
+
+    await SystemChrome.setEnabledSystemUIMode(SystemUiMode.leanBack);
+
+    expect(log, hasLength(1));
+    expect(log.single, isMethodCall(
+      'SystemChrome.setEnabledSystemUIMode',
+      arguments: 'SystemUiMode.leanBack',
+    ));
+  });
+
+  test('setEnabledSystemUIMode asserts for overlays in manual configuration', () async {
```

Comment:
Maybe also add a control test for using manual configuration and correctly passing in a set of overlays